app/src/drawable/3d/box_span/span3d.ts

```diff
@@ -0,0 +1,180 @@
+import * as THREE from "three"
+
+import { SpanPoint3D } from "./spanPoint3d"
+import { SpanLine3D } from "./spanLine3d"
+import { SpanRect3D } from "./spanRect3d"
+import { SpanCuboid3D } from "./spanCuboid3d"
+import { Vector3D } from "../../../math/vector3d"
+
+/**
+ * ThreeJS class for rendering 3D span object
+ */
+export class Span3D {
+  /** First point */
+  private _p1: SpanPoint3D | null
+  /** Second point */
+  private _p2: SpanPoint3D | null
+  /** Third point */
+  private _p3: SpanPoint3D | null
+  /** Fourth point */
+  private _p4: SpanPoint3D | null
+  /** Temporary point that follows mouse cursor */
+  private _pTmp: SpanPoint3D
+  /** Line between first and second point */
+  private _line: SpanLine3D | null
+  /** Cuboid formed by the four points */
+  private _cuboid: SpanCuboid3D | null
+  /** Whether span box is complete */
+  private _complete: boolean
+
+  /** Constructor */
+  constructor() {
+    this._p1 = null
```

Comment:
we can use a list: SpanPoint3D[] to store the points. this can make the code cleaner. Use its length to differentiate different situations

Reply:
Fixed in commit 3960bd5